ResearchKit/ActiveTasks/ORKGoNoGoViewController.m

```diff
@@ -0,0 +1,328 @@
+ this list of conditions and the following disclaimer in the documentation and/or
+ other materials provided with the distribution.
+ 
+ 3.  Neither the name of the copyright holder(s) nor the names of any contributors
+ may be used to endorse or promote products derived from this software without
+ specific prior written permission. No license is granted to the trademarks of
+ the copyright holders even if such marks are included in this software.
+ 
+ THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE
+ FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+
+#import "ORKGoNoGoViewController.h"
+
+#import "ORKActiveStepView.h"
+#import "ORKGoNoGoContentView.h"
+
+#import "ORKActiveStepViewController_Internal.h"
+
+#import "ORKGoNoGoStep.h"
+#import "ORKResult.h"
+
+#import "ORKHelpers_Internal.h"
+
+#import <AudioToolbox/AudioServices.h>
+#import <CoreMotion/CMDeviceMotion.h>
+
+
+@implementation ORKGoNoGoViewController {
+    ORKGoNoGoContentView *_gonogoContentView;
+    
+    NSMutableArray *_results;
+    NSTimer *_stimulusTimer;
+    NSTimer *_timeoutTimer;
+    NSTimeInterval _stimulusTimestamp;
+    BOOL _validResult;
+    BOOL _timedOut;
+    BOOL _shouldIndicateFailure;
+    NSMutableArray<NSNumber*>* tests;
+    BOOL go;
+}
+
+static const NSTimeInterval OutcomeAnimationDuration = 0.3;
+
+#pragma mark - UIViewController
+
+- (void)viewDidLoad {
+    [super viewDidLoad];
+    // Do any additional setup after loading the view.
+    [self configureTitle];
+    _results = [NSMutableArray new];
+    go = true;
+    UIColor* color = self.view.tintColor;
+    _gonogoContentView = [[ORKGoNoGoContentView alloc] initWithColor:color];
+    
+    // Generate the type of tests we are going to display
+    // Always do go first, and make sure there is at least 1 no-go
+    tests = [NSMutableArray array];
+    [tests addObject:[NSNumber numberWithBool:YES]];
+    while (tests.count < [self gonogoTimeStep].numberOfAttempts)
+        [tests addObject:[NSNumber numberWithBool:((float)arc4random_uniform(RAND_MAX) / RAND_MAX) < 0.667]];
+    
+    // Check to make sure we have a no go
+    BOOL hasNoGo = NO;
+    for (NSNumber* go in tests)
+        if ([go boolValue] == NO)
+            hasNoGo = YES;
+    
+    // If not, put one in
+    if (!hasNoGo && tests.count > 1)
```

Comment:
Please add curly-brackets. This is the kind of thing that can get clobbered by a git merge.